doc/specs/#2563 - closeOnExit and TerminalConnection evolution.md

```diff
@@ -0,0 +1,79 @@
+---
+author: Dustin Howett @DHowett-MSFT
+created on: 2019-07-19
+last updated: 2019-11-05
+issue id: "#2563"
+---
+
+# Improvements to CloseOnExit
+
+## Abstract
+
+This spec describes an improvement to the `closeOnExit` profile feature and the `TerminalConnection` interface that will offer greater flexibility and allow us to provide saner defaults.
+
+## Inspiration
+
+Other terminal emulators like ConEmu have a similar feature.
+
+## Solution Design
+
+1. The existing `closeOnExit` profile key will become an enumerated string key with the following values/behaviours:
+   * `always` - a tab or pane hosting this profile will always be closed when the launched connection terminates.
+   * `graceful` - a tab or pane hosting this profile will be closed **iff** the launched connection closes with an exit code == 0.
```

Comment:
I'm guessing an exit code of 0 means that it was expected to close?

So if an unexpected error/failure occurs, the pane will remain open?

Say I set the `commandline` setting to a faulty `.bat` file that decides to fail, does that mean that the pane will remain open with the details of why my `.bat` file failed?

Reply:
Correct.